src/updateConfig.ts

```diff
@@ -0,0 +1,255 @@
+        for (const file of this.files) {
+            if (ignoreFile(file, this.rootDir, this.program)) {
+                continue;
+            }
+            linter.lint(file.fileName, file.text, config);
+        }
+        return linter.getResult();
+    }
+
+    updateConfig(config: Config.RawConfigFile): void {
+        fs.writeFileSync(
+            path.join(this.rootDir, "tslint.json"),
+            stringify(config, { space: 4 }),
+            { encoding: "utf8", flag: "w" });
+    }
+}
+
+function ignoreFile(file: ts.SourceFile, dirPath: string, program: ts.Program): boolean {
+    return program.isSourceFileDefaultLibrary(file) || isExternalDependency(file, path.resolve(dirPath), program);
+}
+
+function walkPackageDir(rootDir: string): LintPackage[] {
+    const packages: LintPackage[] = [];
+
+    function walk(curPackage: LintPackage, dir: string): void {
+        for (const ent of fs.readdirSync(dir, { encoding: "utf8", withFileTypes: true })) {
+            const entPath = path.join(dir, ent.name);
+            if (ent.isFile()) {
+                curPackage.addFile(entPath);
+            } else if (ent.isDirectory() && ent.name !== "node_modules") {
+                if (isVersionDir(ent.name)) {
+                    const newPackage = new LintPackage(entPath);
+                    packages.push(newPackage);
+                    walk(newPackage, entPath);
+                } else {
+                    walk(curPackage, entPath);
+                }
+            }
+        }
+    }
+
+    const lintPackage = new LintPackage(rootDir);
+    packages.push(lintPackage);
+    walk(lintPackage, rootDir);
+    return packages;
+}
+
+/**
+ * Returns true if directory name matches a TypeScript or package version directory.
+ * Examples: `ts3.5`, `v11`, `v0.6` are all version names.
+ */
+function isVersionDir(dirName: string): boolean {
+    return /^ts\d+\.\d$/.test(dirName) || /^v\d+(\.\d+)?$/.test(dirName);
+}
+
+type RuleOptions = boolean | unknown[];
+type RuleDisabler = (failures: IRuleFailureJson[]) => RuleOptions;
+const defaultDisabler: RuleDisabler = _ => {
+    return false;
+};
+const ruleDisablers: Map<string, RuleDisabler> = new Map([
+    ["npm-naming", npmNamingDisabler],
+]);
+
+function disableRules(allFailures: RuleFailure[]): Config.RawRulesConfig {
+    const ruleToFailures: Map<string, IRuleFailureJson[]> = new Map();
+    for (const failure of allFailures) {
+        const failureJson = failure.toJson();
+        if (ruleToFailures.has(failureJson.ruleName)) {
+            ruleToFailures.get(failureJson.ruleName)!.push(failureJson);
+        } else {
+            ruleToFailures.set(failureJson.ruleName, [failureJson]);
+        }
+    }
+
+    const newRulesConfig: Config.RawRulesConfig = {};
+    ruleToFailures.forEach((failures, rule) => {
+        if (ignoredRules.includes(rule)) {
+            return;
+        }
+        const disabler = ruleDisablers.get(rule) || defaultDisabler;
```

Comment:
I would personally hard-code this with a conditional instead of a Map+default since there's only one complex rule right now.